tests/unit/deferred.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2016 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+'''Functional test for deferred.py'''
+
+import pytest
+import os
+import tempfile
+
+from hil import config, deferred, model
+from hil.model import db, Switch
+from hil.test_common import config_testsuite, config_merge, \
+                             fresh_database, fail_on_log_warnings
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+fail_on_log_warnings = pytest.fixture(autouse=True)(fail_on_log_warnings)
+fresh_database = pytest.fixture(fresh_database)
+
+INTERFACE1 = '104/0/10'
+INTERFACE2 = '104/0/18'
+
+DeferredTestSwitch = None
+temp_db = tempfile.NamedTemporaryFile()
+
+
+@pytest.fixture
+def configure():
+    config_testsuite()
+
+    additional_config = {
+        'extensions': {
+            'hil.ext.obm.mock': ''
+            }
+        }
+
+    # if we are using sqlite's in memory db, then change uri to a db on disk
+    uri = config.cfg.get('database', 'uri')
+    if uri == 'sqlite:///:memory:':
+        uri = 'sqlite:///' + temp_db.name
```

Comment:
One way around that is to just do a yield in the else branch as well; this way the fixture machinery still does the right thing, even if there isn't actually any cleanup to do.

The bigger issue is that by initializing temp_db at global scope, all the tests are sharing that one file, as opposed to having their own. As @henn mentioned, this will likely cause problems for parallelizing the tests. I suspect those problems will just be that the tests will have to acquire a lock on the db, and thus serialize, which is why they're still passing, but that's still sub-optimal.